pkg/data/accounts.go

```diff
@@ -0,0 +1,96 @@
+package data
+
+import (
+	"fmt"
+
+	"github.com/Optum/dce/pkg/errors"
+	"github.com/Optum/dce/pkg/model"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/aws/aws-sdk-go/service/dynamodb/expression"
+)
+
+// queryAccounts for doing a query against dynamodb
+func (a *Account) queryAccounts(q *model.Account, keyName string, index string) (*model.Accounts, error) {
+	var expr expression.Expression
+	var bldr expression.Builder
+	var err error
+	var res *dynamodb.QueryOutput
+
+	keyCondition, filters := getFiltersFromStruct(q, &keyName)
+	bldr = expression.NewBuilder().WithKeyCondition(*keyCondition)
+	if filters != nil {
+		bldr = bldr.WithFilter(*filters)
+	}
+
+	expr, err = bldr.Build()
+	if err != nil {
+		return nil, errors.NewInternalServer("unable to build query", err)
+	}
+
+	res, err = a.DynamoDB.Query(&dynamodb.QueryInput{
+		TableName:                 aws.String(a.TableName),
+		IndexName:                 aws.String(index),
+		KeyConditionExpression:    expr.KeyCondition(),
+		ConsistentRead:            aws.Bool(a.ConsistentRead),
+		FilterExpression:          expr.Filter(),
+		ExpressionAttributeNames:  expr.Names(),
+		ExpressionAttributeValues: expr.Values(),
+	})
+
+	if err != nil {
+		return nil, errors.NewInternalServer(
+			fmt.Sprintf("failed to get accounts by status %q", *q.Status),
```

Comment:
How do we know it's get account by status is the filter used?

Reply:
Good catch.  Corrected.